assets/js/themes.js

```diff
@@ -33,3 +36,14 @@ function set_mode(bool) {
         toggle_theme.children[0].setAttribute('class', 'icon ion-ios-moon');
     }
 }
+
+function update_mode (mode) {
+    set_mode(
+           mode == 'true' // for backwards compatibility
+        || mode == 'dark'
+        || (mode != 'light'
+            && window.matchMedia('(prefers-color-scheme: dark)').matches)
```

Comment:
Mind making this a little more explicit? Maybe something like:
```javascript
if (mode == 'true') {
  set_mode(true)
} else if (mode == 'dark') {
  set_mode(true)
) // ...
```

Currently I'm having a bit of trouble understanding when each mode is set.

Reply:
Sure, I'll do that. Sorry for the late response.

Reply:
No worries!